### PR TITLE
Introduce code splitting

### PR DESCRIPTION
**Pull request by marialani**
## What does this change?
This PR introduces route based code splitting using React Lazy and Suspense. A Suspense fallback component has also been added which acts as an intermediary view whilst each separate code chunk is loading. React Lazy works with files that have a default export so the relevant pages have been updated.

The tsconfig `module` value has been changed from `commonjs` to `ESnext` in order to get the code splitting to work as expected, with the previous setting webpack had trouble separating chunks. As a result of changing the tsconfig the jest configuration has also been updated to ensure it uses commonjs.

## How can we measure success?
- the parsed bundle size was 1.32mb and is now 1.06mb
- the gzipped bundle size was 362kb and is now 303.19kb

## Have we considered potential risks?
There was a risk of users seeing a blank whilst the page was loading. We have addressed this by including an intermediary view (pageSkeleton) whilst chunks are loading.

## Images
This is what it looks like on a fast 3G connection

Successfully tested on CODE

Co-authored by @ripecosta @rBangay

### CODE REVIEW

File: app/client/components/user.tsx [hunk 1]
```diff
@@ -15,181 +15,197 @@ import {
...
+const Settings = lazy(() => import("./identity/Settings"));
+const Help = lazy(() => import("./help"));
+const HelpCentre = lazy(() => import("./helpCentre/helpCentre"));
+const ContactUs = lazy(() => import("./contactUs/contactUs"));
+
 const User = () => (
   <Main>
     <Global styles={css(`${global}`)} />
     <Global styles={css(`${fonts}`)} />
+    {/* replace with a real fallback component */}
```
Comment (rBangay) [inline app/client/components/user.tsx:62]: we can probably remove this comment

File: app/client/components/pageSkeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        )[0];
+
+        if (selectedNonMMALocationObject) {
+          return (
+            <>
+              <SectionHeader title={selectedNonMMALocationObject.title} />
+              <SectionPageContainer sectionTitle="&nbsp;">
+                <div
+                  css={css`
+                    margin-bottom: ${space[24]}px;
```
Comment (arausuy) [inline app/client/components/pageSkeleton.tsx:147]: How safe is this access by index?

File: app/client/components/user.tsx (hunk 1 above)
Comment (arausuy) [reply app/client/components/user.tsx:62]: 👍

File: app/client/components/pageSkeleton.tsx (hunk 2 above)
Comment (marialani) [reply app/client/components/pageSkeleton.tsx:147]: I believe there will be some refactoring done here once there is a bit more standardisation across the codebase. This works for now with what we have, on safety I will defer to @rBangay and @ripecosta.

**Review by arausuy — APPROVED** [review]

File: app/client/components/pageSkeleton.tsx (hunk 2 above)
Comment (ripecosta) [reply app/client/components/pageSkeleton.tsx:147]: Very safe. It is made available by the Guardian Design System (Source components). Documentation for `space` [here](https://www.theguardian.design/2a1e5182b/p/449bd5-space)

**Review by ripecosta — APPROVED** [review]
Re-Approving, following rebase!

**Pull request merged**